src/store.js

```diff
@@ -0,0 +1,29 @@
+import {createStore, applyMiddleware, compose} from 'redux';
+import rootReducer from './reducers';
+import ReduxThunk from 'redux-thunk/';
+import {persistStore, persistReducer} from 'redux-persist';
+import storage from 'redux-persist/lib/storage';
+import autoMergeLevel1 from 'redux-persist/lib/stateReconciler/autoMergeLevel1';
+
+const persistConfig = {
+    key: 'root',
+    storage: storage,
+    // autoMergeLevel1 is the default anyway, but I want to be explicit
+    stateReconciler: autoMergeLevel1,
+    whitelist: ['auth', 'map']
+};
+
+const persistedReducer = persistReducer(persistConfig, rootReducer);
+
+export const store = createStore(
+    persistedReducer,
+    {},
+    compose(
+        applyMiddleware(ReduxThunk)
+    ));
```

Comment:
Finally to create our `store` that we use in `App.js`, instead of passing in our `rootReducer` we now pass in the `persistedReducer` which handles the in-flight reducers of the app same as `rootReducer` would have, but it also handles the persistence on the fly.